payment/key.go

```diff
@@ -77,3 +83,65 @@ func GenerateSecret() (secret string, nonce string, err error) {
 
 	return fmt.Sprintf("%x", encryptedBytes), fmt.Sprintf("%x", nonceBytes), err
 }
+
+// LookupPublicKey returns the merchant key corresponding to the keyID used for verifying requests
+func (service *Service) LookupPublicKey(ctx context.Context, keyID string) (*httpsignature.Verifier, error) {
+	rootKeyIDStr, caveats, err := cryptography.DecodeKeyID(keyID)
+	if err != nil {
+		return nil, err
+	}
+
+	rootKeyID, err := uuid.FromString(rootKeyIDStr)
+	if err != nil {
+		return nil, fmt.Errorf("root key id must be a uuid: %v", err)
+	}
+
+	key, err := service.Datastore.GetKey(rootKeyID, false)
+	if err != nil {
+		return nil, err
+	}
+
+	secretKeyStr := key.SecretKey
+
+	if caveats != nil {
+		_, secretKeyStr, err = cryptography.Attenuate(rootKeyID.String(), "secret-token:"+secretKeyStr, caveats)
```

Comment:
should we just start using `secret-token:...` syntax in the secret key instead of this append?

Reply:
I didn't want to break compat for existing keys - but thinking about it more, since we don't actually use these for verifying the macaroons it doesn't matter much. will change

Reply:
:+1:  now is the time to do this, rather than later imo.